app/services/licences/supplementary/process-imported-licence.service.js

```diff
@@ -0,0 +1,125 @@
+
+  await PersistSupplementaryBillingFlagsService.go(
+    twoPartTariffBillingYears,
+    flagForPreSrocSupplementary,
+    flagForSrocSupplementary,
+    licenceId
+  )
+}
+
+function _earliestChangedDate (importedLicence, existingLicenceDetails) {
+  const { endDate: currentFinancialYearEndDate } = determineCurrentFinancialYear()
+  const changedDates = []
+
+  let date
+
+  // NOTE: Because checking if dates are equal does not give the expected result in JavaScript (it sees 2 different
+  // objects even if they represent the same date and time) we convert the date object to strings allowing us to compare
+  // the two
+  if (String(importedLicence.expiredDate) !== String(existingLicenceDetails.expired_date)) {
+    date = importedLicence.expiredDate ?? existingLicenceDetails.expired_date
+    changedDates.push(date)
+  }
+
+  if (String(importedLicence.lapsedDate) !== String(existingLicenceDetails.lapsed_date)) {
+    date = importedLicence.lapsedDate ?? existingLicenceDetails.lapsed_date
+    changedDates.push(date)
+  }
+
+  if (String(importedLicence.revokedDate) !== String(existingLicenceDetails.revoked_date)) {
+    date = importedLicence.revokedDate ?? existingLicenceDetails.revoked_date
+    changedDates.push(date)
+  }
+
+  // Filter out those greater than the current financial year end date
+  const filteredDates = changedDates.filter((changedDate) => {
+    return (changedDate < currentFinancialYearEndDate)
+  })
+
+  // Now work out the earliest end date from those that have changed
+  return filteredDates.length > 0 ? new Date(Math.min(...filteredDates)) : null
+}
+
+function _flagForPresrocSupplementary (existingLicenceDetails, earliestChangedDate) {
+  const { flagged_for_presroc: flagged, pre_sroc_charge_versions: chargeVersions } = existingLicenceDetails
+
+  // If the licence has no presroc charge versions return false. We check this before the existing flag, because this
+  // is an opportunity to remove the PRESROC supplementary billing flag from a licence that won't ever be picked up by
+  // the PRESROC billing engine (so the flag will never get removed)
+  if (!chargeVersions) {
+    return false
+  }
+
+  // If it is already flagged then we don't want to change the flag so return true
+  if (flagged) {
+    return true
+  }
+
+  // Else return the result of checking if the earliest changed date was before PRESROC billing regime ended
+  return earliestChangedDate < SROC_START_DATE
+}
+
+function _flagForSrocSupplementary (existingLicenceDetails) {
+  const { sroc_charge_versions: chargeVersions } = existingLicenceDetails
+
+  // If the licence has no sroc charge versions return false. We check this before the existing flag, because this
+  // is an opportunity to remove the SROC supplementary billing flag from a licence that won't ever be picked up by
+  // the SROC billing engine (so the flag will never get removed)
+
+  return !!chargeVersions
```

Comment:
Good call on removing my unnecessary check of the existing flag! But it does mean the comment needs to be updated accordingly.